src/web/WhitelistHandler.h

```diff
@@ -159,17 +163,27 @@ class WhitelistHandler {
     }
 
 private:
+    template <SomeResolver HostnameResolverType>
     [[nodiscard]] static std::unordered_set<std::string>
-    getWhitelist(util::Config const& config)
+    getWhitelist(util::Config const& config, HostnameResolverType&& resolver)
     {
-        using SetType = std::unordered_set<std::string> const;
-
         auto whitelist = config.arrayOr("dos_guard.whitelist", {});
         auto const transform = [](auto const& elem) { return elem.template value<std::string>(); };
 
-        return SetType{
+        std::unordered_set<std::string> const hostnames{
             boost::transform_iterator(std::begin(whitelist), transform),
             boost::transform_iterator(std::end(whitelist), transform)};
+
+        // resolve hostnames to ips
+        std::unordered_set<std::string> ips;
+        for (auto const& hostname : hostnames) {
```

Comment:
Does this code work fine with entries that are in CIDR format (e.g. `182.168.0.1/22`)?

Reply:
Yes, because resolver checks whether it is an IP or network address and actually resolves only hostnames. See function `isAddress()` in `Resolver`.
Also I put mock resolver only into one test while two others are using the real one.